### ISSUE TITLE
main.js - TypeError, ipcMain is undefined

### ISSUE THREAD

**Issue by sgornick** [OpenBazaar/OpenBazaar-Client#1509]
Upgraded to Client v1.1.5 and get this error: TypeError: Cannot read property 'on' of undefined

```
$ npm start

> OpenBazaar@1.1.4 start /opt/OpenBazaar-Client-1.1.5
> electron main.js

App threw an error when running [TypeError: Cannot read property 'on' of undefined]
A JavaScript error occurred in the main process
Uncaught Exception:
TypeError: Cannot read property 'on' of undefined
    at Object.<anonymous> (/opt/OpenBazaar-Client-1.1.5/main.js:189:8)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at loadApplicationPackage (/opt/OpenBazaar-Client-1.1.5/node_modules/electron-prebuilt/dist/resources/default_app.asar/main.js:253:23)
    at Object.<anonymous> (/opt/OpenBazaar-Client-1.1.5/node_modules/electron-prebuilt/dist/resources/default_app.asar/main.js:293:5)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)

```

This was after doing
`$ sudo npm install electron-prebuilt`

Ubuntu 14.04.4 LTS

I see there were changes recently for badge counts.
https://github.com/OpenBazaar/OpenBazaar-Client/commit/4e748b0f815705e6d361bcfc16a7b3526059ccbc

From main.js, 
Prior to that commit:
`var ipcMain = require('ipc-main');`

After that commit

```
var ipcMain = require('ipc-main');
var ipcMain = require('electron').ipcMain;
```

(Yes, ipcMain is assigned a second time.)

So it appears that ipcMain is left undefined, and then later the ipcMain.on event handler is causing an error as ipcMain is undefined.

**Comment by hoffmabc** [comment]
Which electron-prebuilt is this on?

**Comment by sgornick** [comment]
> Which electron-prebuilt is this on?

 I didn't save any logs on that, and can no longer tell.

To get from a clean start I wiped Client/node_modules, then removed nodejs and npm, and wiped ~/.npm.  
Then I reinstalled those two packages, installed electron-prebuilt and then did npm install.   OB Client v1.1.5 then ran without the exception any more.    So it was something with my electron installation certainly.

But should there be some better handling in main.js when 
`var ipcMain = require('electron').ipcMain;` ends up undefined due to a bad electron?

[Edit: Incidentally, before trying OB Client v1.1.5, the electron I had was working fine with OB Client v1.1.4.   So it appears that something changed in OB Client 1.1.5 which didn't like my electron configuration.   And the assigning require('electron').ipcMain (from v1.1.5) instead of require('ipc-main'); (as existed in v1.1.4) would seem to be the change that no longer would work with my electron that I had.]
